include/lib/utils_def.h

```diff
@@ -57,16 +57,19 @@
 	(((ptr) > UINTPTR_MAX - (inc)) ? 1 : 0)
 
 /*
- * For those constants to be shared between C and other sources, apply a 'ull'
- * suffix to the argument only in C, to avoid undefined or unintended behaviour.
+ * For those constants to be shared between C and other sources, apply a 'ul'
+ * or 'ull' suffix to the argument only in C, to avoid undefined or unintended
+ * behaviour.
  *
- * The GNU assembler and linker do not support the 'ull' suffix (it causes the
- * build process to fail) therefore the suffix is omitted when used in linker
- * scripts and assembler files.
+ * The GNU assembler and linker do not support the 'ul' and 'ull' suffix (it
+ * causes the build process to fail) therefore the suffix is omitted when used
+ * in linker scripts and assembler files.
 */
 #if defined(__LINKER__) || defined(__ASSEMBLY__)
+# define  UL(_x)	(_x)
 # define ULL(_x)	(_x)
 #else
+# define  UL(_x)	(_x##ul)
```

Comment:
We avoid using `long` instead of `long long` because it causes problems when compiling for AArch32. In your platform it doesn't matter because it's only built for AArch64, but I'm not sure about adding this to the generic code.

https://github.com/ARM-software/arm-trusted-firmware/wiki/ARM-Trusted-Firmware-Coding-Guidelines#use-of-built-in-c-and-stdlib-data-types

> Avoid use of `long`. This is guaranteed to be at least 32-bit but given that `int` is 32-bit on ARM platforms, there is no use for it. For integers at least 64-bit, use `long long`.